extensions/integration-tests/src/notebook.test.ts

```diff
@@ -104,9 +104,49 @@ if (context.RunTest) {
 				assert(sparkResult === '2', `Expected spark result: 2, Actual: ${sparkResult}`);
 			});
 		}
+
+		test('scala language test', async function () {
+			let language = 'scala';
+
+			languageTest(pySparkNotebookContent, pySpark3KernelMetadata, this.test.title, language, {
+				name: language,
+				version: '',
+				mimetype: ''
+			});
+		});
+
+		test('cplusplus language test', async function () {
+			let language = 'cplusplus';
+			languageTest(pySparkNotebookContent, pySpark3KernelMetadata, this.test.title, language, {
+				name: language,
+				version: '',
+				mimetype: ''
+			});
+		});
+
+		test('Empty language test', async function () {
+			let language = '';
+			languageTest(pySparkNotebookContent, pySpark3KernelMetadata, this.test.title, language, {
+				name: language,
+				version: '',
+				mimetype: 'x-scala'
+			});
+		});
 	});
 }
 
+async function languageTest(content: azdata.nb.INotebookContents, kernelMetadata: any, testName: string, languageConfigured: string, languageInfo?: any) {
```

Comment:
nit: Please be more descriptive with method names and/or give it docs. Ideally one shouldn't have to read the entire method to have a general understanding of what the method is doing. Something like 

'verifyNotebookLoadsAndDisplaysLanguage'